### PR TITLE
En_Vm

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 1]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+    if ((this->unk_216 < 0xAAA) || (Player_GetMask(globalCtx) == PLAYER_MASK_STONE) || (this->unk_214 == 0)) {
+        this->unk_214 = 0;
+        if (Math_StepToF(&this->unk_220, 0.0f, 0.003f)) {
+            this->unk_210 = 0;
+            this->unk_224 = 0.0f;
+            func_808CC420(this);
+        }
+    } else {
+        this->unk_214--;
+        if (this->unk_214 < 301) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:306]:
```suggestion
        if (this->unk_214 <= 300) {
```
probably will make more sense.

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 2]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+        if (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.focus.pos, &this->unk_234, &sp5C, &sp4C, true,
+                                    true, false, true, &sp48)) {
+            this->unk_224 = Math_Vec3f_DistXYZ(&this->actor.focus.pos, &sp5C) - 5.0f;
+            this->unk_210 = 2;
+            Math_Vec3f_Copy(&this->unk_234, &sp5C);
+        } else if (this->unk_210 != 0) {
+            this->unk_210 = 1;
+        }
+
+        sp50.x = (Math_CosS(this->actor.shape.rot.y + this->unk_218) * 5.0f) + this->unk_234.x;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:499]: Can this addition go the other way round to match the z one?

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 3]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+    EnVm* this = THIS;
+
+    Collider_DestroyTris(globalCtx, &this->colliderTris);
+    Collider_DestroyJntSph(globalCtx, &this->colliderJntSph);
+}
+
+void func_808CC420(EnVm* this) {
+    f32 lastFrame = Animation_GetLastFrame(&object_vm_Anim_000068);
+
+    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, 1.0f, lastFrame, lastFrame, 2, 0.0f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:205]:
```suggestion
    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, 1.0f, lastFrame, lastFrame, ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 4]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+            EffectSsDeadDd_Spawn(globalCtx, &this->unk_228, &gZeroVec3f, &gZeroVec3f, &D_808CD5BC, &D_808CD5C0, 150,
+                                 -25, 16, 20);
+            func_808CC788(this);
+        }
+    }
+}
+
+void func_808CC788(EnVm* this) {
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_BIMOS_AIM);
+    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, 3.0f, 3.0f, 7.0f, 2, 0.0f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:270]:
```suggestion
    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, 3.0f, 3.0f, 7.0f, ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 5]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+            if (this->unk_210 == 0) {
+                this->unk_210 = 1;
+            }
+        }
+    }
+}
+
+void func_808CCA10(EnVm* this) {
+    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, -1.0f, Animation_GetLastFrame(&object_vm_Anim_000068),
+                     0.0f, 2, 0.0f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:327]:
```suggestion
    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, -1.0f, Animation_GetLastFrame(&object_vm_Anim_000068),
                     0.0f, ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Vm/z_en_vm.c [hunk 6]
```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
...
+            this->skelAnime.playSpeed = 2.0f;
+        } else {
+            func_808CC420(this);
+        }
+    }
+}
+
+void func_808CCBE4(EnVm* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, -1.0f, Animation_GetLastFrame(&object_vm_Anim_000068),
+                     0.0f, 2, 0.0f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Vm/z_en_vm.c:370]:
```suggestion
    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, -1.0f, Animation_GetLastFrame(&object_vm_Anim_000068),
                     0.0f, ANIMMODE_ONCE, 0.0f);
```

**Review by hensldm — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — CHANGES_REQUESTED** [review]
See @hensldm's suggestions

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**